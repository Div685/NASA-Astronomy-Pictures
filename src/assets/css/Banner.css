.banner {
  color: white;
  object-fit: contain;
  height: 424px;
}

.banner__contents {
  margin-left: 30px;
  padding-top: 140px;
  height: 306px;
}

.banner__title {
  font-size: 3rem;
  font-weight: 800;
  padding-bottom: 0.3rem;
  display: flex;
  justify-content: center;
  align-items: center;
}

.banner__fadeBottom {
  height: 7.4rem;
  background-image:
    linear-gradient(
      180deg,
      transparent,
      rgba(37, 37, 37, 0.61),
      #1a1a29
    );
}
