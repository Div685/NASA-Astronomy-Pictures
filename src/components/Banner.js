import React from 'react';
import '../assets/css/Banner.css';

function Banner() {
  return (
    <header
      className="banner"
      style={{
        backgroundSize: 'cover',
        backgroundImage: 'url("https://apod.nasa.gov/apod/image/1004/NGC1055_crawford950.jpg")',
        backgroundPosition: 'center',
      }}
    >
      <div className="banner__contents">
        <h1 className="banner__title">
          Welcome to AstroSplash
        </h1>
      </div>
      <div className="banner__fadeBottom" />

    </header>
  );
}

export default Banner;
