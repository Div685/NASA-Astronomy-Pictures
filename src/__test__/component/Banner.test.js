import React from 'react';
import renderer from 'react-test-renderer';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import Banner from '../../components/Banner';

describe('Routes Component', () => {
  it('renders Routes page correctly', () => {
    const tree = renderer.create(<Banner />);
    expect(tree).toMatchSnapshot();
  });
});

describe('Test Routes Component for NavBar Links', () => {
  it('it test the home page', () => {
    render(<Banner />);
    expect(screen.getByRole('heading')).toHaveTextContent('Welcome to AstroSplash');
  });
});
