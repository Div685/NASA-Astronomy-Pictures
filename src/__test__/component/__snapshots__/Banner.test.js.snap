// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Routes Component renders Routes page correctly 1`] = `
<header
  className="banner"
  style={
    Object {
      "backgroundImage": "url(\\"https://apod.nasa.gov/apod/image/1004/NGC1055_crawford950.jpg\\")",
      "backgroundPosition": "center",
      "backgroundSize": "cover",
    }
  }
>
  <div
    className="banner__contents"
  >
    <h1
      className="banner__title"
    >
      Welcome to AstroSplash
    </h1>
  </div>
  <div
    className="banner__fadeBottom"
  />
</header>
`;
